performance-tests/bench_2/builder/Participant.cpp

```diff
@@ -19,8 +19,9 @@ Participant::Participant(const ParticipantConfig& config, ParticipantReport& rep
   Log::log() << "Creating participant: '" << name_ << "' in domain " << domain_ << std::endl;
 
   // Customize QoS Object
+  DDS::DomainParticipantFactory_var dpf = TheParticipantFactory;
   DDS::DomainParticipantQos qos;
-  TheParticipantFactory->get_default_participant_qos(qos);
+  dpf->get_default_participant_qos(qos);
```

Comment:
I wish this wasn't necessary or at least the type system helped enforce this by returning something like a `shared_ptr`.

Reply:
Eventually we'll do just that...

Reply:
I don't have the code in front of me... but presumably this is returning something allocated that I should have been trapping in a scoped wrapper?